examples/playground/index.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import ReactDOMServer from 'react-dom/server';
+
+import { useLazyPhase, MODE } from 'react-loosely-lazy';
+import { listeners } from './utils';
+
+import {
+  ForPaintComponents,
+  AfterPaintComponents,
+  LazyComponents,
+  CustomWaitComponents,
+  buildServerComponents,
+  buildClientComponents,
+} from './components';
+
+const steps = [
+  'SSR',
+  'PAINT LOADING',
+  'PAINT FETCHING',
+  'PAINT READY',
+  'AFTER LOADING',
+  'AFTER FETCHING',
+  'AFTER READY', // also 'LAZY READY'
+  'CUSTOM LOADING',
+  'CUSTOM FETCHING',
+  'CUSTOM READY',
+];
+
+/**
+ * Controls App
+ */
+const Controls = () => {
+  const [stepIndex, setStepIndex] = useState(0);
+  const [step, setStep] = useState(steps[stepIndex]);
+  const resetSteps = () => setStepIndex(0);
+
+  useEffect(() => {
+    (window as any).step = steps[stepIndex];
```

Comment:
Would it be possible / feasible to convert this to a shared ref under the same react tree?